### DISCUSSION TITLE
No endpoints could be reached

### DISCUSSION THREAD

**Discussion by MikhailTushev** [rabbitmq/rabbitmq-stream-dotnet-client#326] [Q&A]
Hello everyone!)
I have a problem like "could not reach the endpoints".
I use test-container to test the logical with stream, everything works well in the local, but in the server the test failed with the same error: it was not possible to reach the endpoints.
It is set up this way:
![image](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/assets/71391302/d28ff3ae-58cf-48fd-ab57-45af38a7bae3)

i tried connect like that
![image](https://github.com/rabbitmq/rabbitmq-stream-dotnet-client/assets/71391302/f485b08c-7be8-416e-a5f6-8d804473fd10)

what need to do for success connect? (advertising host tried setup like "someHost")

image of rabbitMQ - 3.11

**Comment by Gsantomaggio** [comment]
The client has to lookup the IP address.  Try to ping the IP address or `nslookup` the hostname

**Reply by MikhailTushev** [reply]
does the setup look ok at first glance?

**Reply by Gsantomaggio** [reply]
I never executed rabbitmq in that way. I don't know. but the host looks strange

**Reply by MikhailTushev** [reply]
Hm... I just don't understand how I can connect to rmq streams without a dns hostname and set it to advertising host...  
else way, i can use an address resolver without a proxy server or load balancer.. but when installing replication in rmq, it will fail. hm

**Reply by Gsantomaggio** [reply]
You can connect the client to server using the IP or DNS. The `advertising host` is needed when the client cannot lookup the DNS.
for example:
```
 docker run -it --rm --name rabbitmq-stream-client-test \
                -p 5552:5552 -p 5672:5672 -p 15672:15672 \
                -e RABBITMQ_SERVER_ADDITIONAL_ERL_ARGS="-rabbitmq_stream advertised_host localhost" \
```
The client can't access the docker hostname but it will use `localhost`

**Comment by Gsantomaggio** [comment]
closing! Feel free to reopen with more details
